src/Features/Core/Portable/SemanticSearch/AbstractSemanticSearchService.cs

```diff
@@ -184,12 +187,38 @@ ExecuteQueryResult CreateResult(ImmutableArray<QueryCompilationError> compilatio
         }
     }
 
-    private static bool TryGetFindMethod(Assembly queryAssembly, [NotNullWhen(true)] out MethodInfo? method, out QueryKind queryKind, out string? error, out string[]? errorMessageArgs)
+    private static void SetModuleCancellationToken(Assembly queryAssembly, CancellationToken cancellationToken)
+    {
+        var pidType = queryAssembly.GetType("<PrivateImplementationDetails>", throwOnError: true);
```

Comment:
who ends up potentially catching if this happens?  or should this be impossible as we own the emitting of these assemblies?

Reply:
We own the emit. It'll be reported as NFW.

Reply:
wfm.  but it would be good to doc imo.